src/pybox/base.py

```diff
@@ -14,31 +14,34 @@ class BasePyBox(ABC):
     """A PyBox is an interface to a kernel that can execute code."""
 
     @abstractmethod
-    def run(self, code: str, **kwargs) -> PyBoxOut | None:
+    def run(self, code: str, **kwargs) -> list[PyBoxOut | CodeExecutionError]:
```

Comment:
I am considering whether `data: list[dict]` is more appropriate.

Reply:
`PyBoxOut` could have a list of data, and an error. Also an empty data field sounds more reasonable.